File: go/7-logika.go
package main
import "fmt"

func iloczyn(a, b bool) {
	fmt.Printf("iloczyn %d && %d = %d\n", a, b, a && b)
}

func suma(a, b bool) {
	fmt.Printf("suma %d || %d = %d\n", a, b, a || b)
}

func negacja(a bool) {
	fmt.Printf("negacja !%d  = %d\n", a, !a)
}


func main() {
	zero:=false
	jeden:=true

	iloczyn(zero, zero)
	iloczyn(zero, jeden)
	iloczyn(jeden, zero)
	iloczyn(jeden, jeden)

	suma(zero, zero)
	suma(zero, jeden)
	suma(jeden, zero)
	suma(jeden, jeden)
	
	negacja(zero)
	negacja(jeden)

	a:=3
	b:=1
	c:=2
	d:=0b10101010
	e:=0b01010101
	
	fmt.Printf("iloczyn: %d & %d = %d\n", a, b, a & b)
	fmt.Printf("suma: %d | %d = %d\n", b, c, b | c)
	fmt.Printf("suma modulo 2 (xor) %d ^ %d = %d\n", b, c, b ^ c)
	fmt.Printf("negacja ^ %d = %d\n", b, ^b & 1 )
	fmt.Printf("negacja ^ %b = %08b\n", d, (^d) & 0xff )
	fmt.Printf("d=%08b e=%08b\n", d, e )

}

File: perl/7-logika.pl
#!/bin/perl

sub iloczyn
{
	$a=@_[0];
	$b=@_[1];
	print "iloczyn $a && $b =",  ($a && $b);
	print "\n";
	print "iloczyn $a and $b =", ($a and $b);
	print "\n";
}

sub suma
{
	$a=@_[0];
	$b=@_[1];
	
	print "suma $a or $b =",  ($a or $b);
	print "\n";
	print "suma $a || $b =",  ($a || $b);
	print "\n";
}

sub pokaz_bool
{
	 if (@_[1]) {
		print "@_[0] to true\n";
	}
	else {
		print "@_[0] to false\n";
	}

}

sub negacja
{
	$a=@_[0];
	$b=(not $a);
	$c=($b and $b);
	$d=$b and $b;
	
	pokaz_bool("a", $a);
	pokaz_bool("b", $b);
	pokaz_bool("c", $c);
	pokaz_bool("d", $d);
	pokaz_bool("negacja $a =", $b);
	
}


$zero=0;
$jeden=1;

iloczyn($zero, $zero);
iloczyn($zero, $jeden);
iloczyn($jeden, $zero);
iloczyn($jeden, $jeden);

suma($zero, $zero);
suma($zero, $jeden);
suma($jeden, $zero);
suma($jeden, $jeden);

negacja($zero);
negacja($jeden);

$a=3;
$b=1;
$c=2;

print "iloczyn: $a & $b =", $a & $b, "\n";
print "suma: $a | $b =", $a | $b, "\n";
print "suma modulo 2: $a ^ $b =", $a ^ $b, "\n";
print "negacja ~$b =", (~b) & 255 ,"\n";

